### DISCUSSION TITLE
Reading a switch/button from ESP32-WROOM with NodeRed

### DISCUSSION THREAD

**Discussion by Eldkatten** [arendst/Tasmota#17610] [General]
Hello,

sorry, this is probably a very basic question, but for the life of me I can't find an answer after two days of "deep search".

My setup is this:
- ESP32-WROOM with Tasmota, named "Generalswitch"
- Node-Red on a Raspberry Pi

For testing:
I configured GPIO12 as "Relay", and I can toggle the state with "cmnd/Generalswitch/power", and read the status with "stat/Generalswitch/POWER" (somehow it only seem to work with all caps).
I configured GPIO35 as "ADC Input", and I can read the value with "tele/Generalswitch/SENSOR".

What I _actually_ want to do is to have just a button or switch, so I configured a few GPIOs as "Input", "Switch", "Button" and "Button tc", but I never get any message, neither with "tele/Generalswitch/#" nor "state/Generalswitch/#".

What am I missing here? Having a simple button or switch should be absolute basics, but here I am failing.

Thank you for any hints or suggestions.

**Comment by Jason2866** [comment]
Read this https://tasmota.github.io/docs/Buttons-and-Switches/
At the top the most important info

**Reply by Eldkatten** [reply]
Thanks a lot :-) Especially for having patience with someone who asks a stupid question ;-)
I was just about to write my own answer, since reading another post here I found out about "rules" and it seems I finally got it working :-)
